setup.py

```diff
@@ -33,7 +33,7 @@
     'install_requires': [
         'pynwb>=1.3.0',
         'numpy<1.19.4',
-        'h5py<3'
+        'h5py',
```

Comment:
@bendichter 
I couldn't install `ndx-photometry` with `h5py<3`, had `legacy-install-failure` error for it. 
The version that this installs with is `3.8.0`.